services/repository/files/cherry_pick.go

```diff
@@ -66,7 +67,7 @@ func CherryPick(ctx context.Context, repo *repo_model.Repository, doer *user_mod
 	}
 	parent, err := commit.ParentID(0)
 	if err != nil {
-		parent = repo.ObjectFormat.EmptyTree()
+		parent = git.ObjectFormatFromName(repo.ObjectFormatName).EmptyTree()
```

Comment:
I still think `ObjectFormatFromName` should default to "sha1", just like git.

In case `repo.ObjectFormatName` is empty.

Reply:
Fixed.